index.js

```diff
@@ -193,11 +216,30 @@ SubresourceIntegrityPlugin.prototype.apply = function apply(compiler) {
       pluginArgs.body.filter(filterTag).forEach(processTag);
       callback(null, pluginArgs);
     }
+
+    /*  Add jsIntegrity and cssIntegrity properties to pluginArgs, to
+     *  go along with js and css properties.  These are later
+     *  accessible on `htmlWebpackPlugin.files`.
+     */
+    function beforeHtmlGeneration(pluginArgs, callback) {
+      ['js', 'css'].forEach(function addIntegrity(fileType) {
+        pluginArgs.assets[fileType + 'Integrity'] =
+          pluginArgs.assets[fileType].map(function assetIntegrity(filePath) {
+            var src = hwpSrcRelativeOutputPath(compilation.compiler,
+                                               pluginArgs.plugin,
+                                               filePath);
+            return compilation.assets[src].integrity;
+          });
+      });
+      callback(null);
```

Comment:
This might cause a deprecated warning (not 100% sure) could you change it to `callback(null, pluginArgs)`?